x-pack/test/api_integration/deployment_agnostic/README.md

```diff
@@ -44,17 +44,25 @@ x-pack/test/<my_own_api_integration_folder>
 │  │  ├─ <api_2>
 │  │  │  ├─ <test_2_1>
 │  │  │  ├─ <test_2_2>
-│  ├─ services
-│  │  ├─ index.ts // only services from 'x-pack/test/api_integration/deployment_agnostic/services'
-│  │  ├─ <deployment_agnostic_service_1>.ts
-│  │  ├─ <deployment_agnostic_service_2>.ts
+│  ├─ configs
+│  │  ├─ stateful
+│  │  │  ├─ <stateful>.index.ts  // e.g., oblt.index.ts
+│  │  │  ├─ <stateful>.config.ts // e.g., oblt.stateful.config.ts
+│  │  ├─ serverless
+│  │     ├─ <serverless_project>.index.ts             // e.g., oblt.index.ts
+│  │     ├─ <serverless_project>.serverless.config.ts // e.g., oblt.serverless.config.ts
 │  ├─ ftr_provider_context.d.ts  // with types of services from './services'
-├─ stateful.index.ts
-├─ stateful.config.ts 
-├─ <serverless_project>.index.ts                // e.g., oblt.index.ts
-├─ <serverless_project>.serverless.config.ts    // e.g., oblt.serverless.config.ts
+│  ├─ services
+│     ├─ index.ts // only services from 'x-pack/test/api_integration/deployment_agnostic/services'
+│     ├─ <deployment_agnostic_service_1>.ts
+│     ├─ <deployment_agnostic_service_2>.ts
 ```
 
+## Loading Your Tests Properly
+When Platform teams add deployment-agnostic tests, it is expected that these tests are loaded in `configs/stateful/platform.index.ts` and at least one of the `<serverless_project>.serverless.config` files under `configs/serverless` folder.
```

Comment:
I think over time, we'll have multiple platform stateful config files, each loading only a subset of the platform tests (for test run time / group balancing reasons). Not sure if we already want to prepare the structure and advice for this now. Might be a good opportunity. What do you think?

Reply:
I was thinking about it, but couldn't decide if we need config for each plugin - could be too much, ES/Kibana restart takes ~1 min for stateful and most of test are run within _few_ seconds (but there are some long ones too). The more tests migrated to deployment agnostic, the more CI minutes we need and I think restarting servers should be taken serious.

I would keep it for now as is and see how the migration goes. I would prefer us to manage configs to guarantee its compatibility in different environments.

But open for ideas and making a change.

Reply:
Yeah, I think it's fine to leave it as is for now. With more tests migrated to deployment agnostic, we'll come to the point where one config is running for too long and we want to split. But we can deal with this situation when it comes closer. I'm fine with leaving this PR as is for now.